### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by WibbenZ**
Removed trailing spaces
Removed brackets around if statments
Removed extra spaces in spells.xml
Updated the last spells
Updated indentation of three monsters
Updated IP ban script
Updated some legacy functions to userdata
Updated some variable names
Updated mass healing (exura gran mas res) - it now only heals players
Replaced all table.remove and table.insert
Replaced some of the for loops
Cleaned some formulas
Cleaned most? spells
Fixed surpisebag.lua
Moved the last "unknown areas" to the lib file, insted of having huge duplicated arrays.
Added ALOT of missing newlines, most in monsters

### CODE REVIEW

File: src/protocolstatus.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ void ProtocolStatus::onRecvFirstMessage(NetworkMessage& msg)
 			if (requestedInfo & REQUEST_PLAYER_STATUS_INFO) {
 				characterName = msg.getString();
 			}
-			g_dispatcher.addTask(createTask(std::bind(&ProtocolStatus::sendInfo, std::dynamic_pointer_cast<ProtocolStatus>(shared_from_this()), 
+			g_dispatcher.addTask(createTask(std::bind(&ProtocolStatus::sendInfo, std::dynamic_pointer_cast<ProtocolStatus>(shared_from_this()),
```
Comment (djarek) [inline src/protocolstatus.cpp]: While you're at it - change std::dynamic_pointer_cast to std::static_pointer_cast.

**Comment by djarek** [comment]
+1

File: src/protocolstatus.cpp (hunk 1 above)
Comment (WibbenZ) [inline src/protocolstatus.cpp]: At line 69 aswell?
````
				g_dispatcher.addTask(createTask(std::bind(&ProtocolStatus::sendStatusString,
									  std::dynamic_pointer_cast<ProtocolStatus>(shared_from_this()))));
```

Comment (djarek) [inline src/protocolstatus.cpp]: yep

Comment (WibbenZ) [inline src/protocolstatus.cpp]: Good then it's done +1

File: data/npc/lib/npcsystem/npchandler.lua [hunk 2]
```diff
@@ -76,29 +76,29 @@ if NpcHandler == nil then
...
+			[MESSAGE_ONBUY] = "Here you are.",
+			[MESSAGE_BOUGHT] = "Bought |ITEMCOUNT|x |ITEMNAME| for |TOTALCOST| gold.",
+			[MESSAGE_SELL] = "Do you want to sell |ITEMCOUNT| |ITEMNAME| for |TOTALCOST| gold coins?",
+			[MESSAGE_ONSELL] = "Here you are, |TOTALCOST| gold.",
+			[MESSAGE_SOLD] = "Sold |ITEMCOUNT|x |ITEMNAME| for |TOTALCOST| gold.",
+			[MESSAGE_MISSINGMONEY] = "You don't have enough money.",
+			[MESSAGE_NEEDMONEY] = "You don't have enough money.",
+			[MESSAGE_MISSINGITEM] = "You don't have so many.",
+			[MESSAGE_NEEDITEM] = "You do not have this object.",
 			[MESSAGE_NEEDSPACE]	= "You do not have enough capacity.",
```
Comment (Kamenuvol) [inline data/npc/lib/npcsystem/npchandler.lua]: Does not need space here in [MESSAGE_NEEDSPACE]
=P

Comment (WibbenZ) [inline data/npc/lib/npcsystem/npchandler.lua]: Fail haha, tought id spend a few mins but kept finding annoying things :p
Ill make sure to fix it tomorow!

Comment (ranisalt) [inline data/npc/lib/npcsystem/npchandler.lua]: Why remove those? They are there to keep alignment, not as extra useless space.

File: data/actions/scripts/other/bed_modification_kits.lua [hunk 3]
```diff
@@ -33,7 +33,8 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
 		return true
 	end
 
-	for _, bed in pairs(BEDS) do
+	for i = 1, #beds do
```
Comment (ninjalulz) [inline data/actions/scripts/other/bed_modification_kits.lua]: Revert this.

File: data/spells/scripts/party/enchant.lua [hunk 4]
```diff
@@ -1,57 +1,65 @@
...
-		doSendMagicEffect(pos, CONST_ME_POFF)
+	local party = creature:getParty()
+	if not party then
+		creature:sendCancelMessage("No party members in range.")
+		position:sendMagicEffect(CONST_ME_POFF)
+		return false
+	end
+
+	local membersList = party:getMembers()
+	membersList = membersList + party:getLeader()
```
Comment (ninjalulz) [inline data/spells/scripts/party/enchant.lua]: Change this to: ```membersList[#membersList + 1] = party:getLeader()```

File: data/spells/scripts/party/enchant.lua [hunk 5]
```diff
@@ -1,57 +1,65 @@
...
+		position:sendMagicEffect(CONST_ME_POFF)
 		return false
 	end
 
 	local affectedList = {}
-	for _, pid in ipairs(membersList) do
-		if(getDistanceBetween(getCreaturePosition(pid), pos) <= 36) then
-			table.insert(affectedList, pid)
+	for i = 1, #membersList do
+		if getDistanceBetween(membersList[i]:getPosition(), position) <= 36 then
```
Comment (ninjalulz) [inline data/spells/scripts/party/enchant.lua]: Use Position.getDistance here

File: data/spells/scripts/party/enchant.lua [hunk 6]
```diff
@@ -1,57 +1,65 @@
...
+		creature:sendCancelMessage("No party members in range.")
+		position:sendMagicEffect(CONST_ME_POFF)
 		return false
 	end
 
 	local mana = math.ceil((0.9 ^ (tmp - 1) * baseMana) * tmp)
-	if(getPlayerMana(cid) < mana) then
-		doPlayerSendDefaultCancel(cid, RETURNVALUE_NOTENOUGHMANA)
-		doSendMagicEffect(pos, CONST_ME_POFF)
+	if player:getMana() < mana then
```
Comment (ninjalulz) [inline data/spells/scripts/party/enchant.lua]: Change player to creature

File: data/npc/lib/npcsystem/npchandler.lua (hunk 2 above)
Comment (WibbenZ) [inline data/npc/lib/npcsystem/npchandler.lua]: They are not even aligning it, it makes the code look ugly as hell xD

File: data/spells/scripts/party/enchant.lua (hunk 6 above)
Comment (WibbenZ) [inline data/spells/scripts/party/enchant.lua]: Tought I fixed all of those, did them abit to quick and copy pasted the spell codes haha..
But it's only this one that I can see needs to be changed.

File: data/actions/scripts/other/bed_modification_kits.lua [hunk 7]
```diff
@@ -1,23 +1,23 @@
...
 }
 
-local function internalBedTransform(item, targetItem, toPosition, ids)
-	targetItem:transform(ids[1])
+local function internalBedTransform(item, targetItem, tile, toPosition, itemArray)
+	targetItem:transform(itemArray[1])
 	targetItem:getPosition():sendMagicEffect(CONST_ME_POFF)
 
-	Tile(toPosition):getItemByType(ITEM_TYPE_BED):transform(ids[2])
+	tile:getItemByType(ITEM_TYPE_BED):transform(itemArray[2])
```
Comment (ninjalulz) [inline data/actions/scripts/other/bed_modification_kits.lua]: Revert tile to Tile(toPosition) since toPosition changes just before the function is called.

File: data/talkactions/scripts/info.lua [hunk 8]
```diff
@@ -19,14 +19,16 @@ function onSay(player, words, param)
 	player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, "Access: " .. (target:getGroup():getAccess() and "1" or "0"))
 	player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, "Level: " .. target:getLevel())
 	player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, "Magic Level: " .. target:getMagicLevel())
-	player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, "Speed: " .. getCreatureSpeed(player:getId()))
+	player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, "Speed: " .. player:getSpeed())
```
Comment (ninjalulz) [inline data/talkactions/scripts/info.lua]: target:getSpeed()

File: data/talkactions/scripts/owner.lua [hunk 9]
```diff
@@ -7,7 +7,8 @@ function onSay(player, words, param)
 		return false
 	end
 
-	local house = House(getTileHouseInfo(player:getPosition()))
+	local tile = Tile(position)
```
Comment (ninjalulz) [inline data/talkactions/scripts/owner.lua]: position is not defined.

File: data/actions/scripts/other/bed_modification_kits.lua (hunk 7 above)
Comment (WibbenZ) [inline data/actions/scripts/other/bed_modification_kits.lua]: Thanks ill fix them right away :)

**Comment by marksamman** [comment]
Please stop piling up changes in the same PR, the more changes you make the harder it is for me and others to review, and the more unlikely it is that this will ever be merged. If you're going to introduce new changes, open a new pull request and keep this strictly to cleanups.

Can you elaborate on why you have changed occurrences of ipairs to access entry by index instead? ipairs is less code, looks much cleaner and is easier to use (harder to screw up because you didn't remember Lua is 1-indexed).

**Comment by WibbenZ** [comment]
Sure, well it's hard to know that you even check the PR when no feedback is given, in that sence it seems easier to just squash them.
This PR is pretty much done, can't find much more that needs to be cleaned.

This is pretty much only cleanups, except surpisebag.lua.

About the loops, to my understanding they are alot quicker (https://springrts.com/wiki/Lua_Performance#TEST_9:_for-loops)

It might look - abit cleaner, but how is it harder to use?
Insted of getting the variable from the loop function, if you want it to be reverted fine.
But as I said, it seems to be alot quicker.

File: data/globalevents/scripts/serversave.lua [hunk 10]
```diff
@@ -15,18 +15,19 @@ local function serverSave()
 	end
 end
 
+local endingText = shutdownAtServerSave and "Please logout." or "Please mind it may freeze!"
 local function secondServerSaveWarning()
-	broadcastMessage("Server is saving game in one minute. Please logout.", MESSAGE_STATUS_WARNING)
+	broadcastMessage(string.format("Server is saving game in one minutes. %s.", endingText), MESSAGE_STATUS_WARNING)
```
Comment (Kamenuvol) [inline data/globalevents/scripts/serversave.lua]: little typo "one minutes"

Comment (marksamman) [inline data/globalevents/scripts/serversave.lua]: This can be reverted altogether since it's inaccurate, every player gets kicked at server save regardless of config.

File: data/actions/scripts/other/bed_modification_kits.lua [hunk 11]
```diff
@@ -1,23 +1,23 @@
...
 	[7905] = {{7819, 7820}, {7821, 7822}}, -- yellow kit
 	[7906] = {{7815, 7816}, {7817, 7818}}, -- red kit
-	[7907] = {{1754, 1755}, {1760, 1761}},  -- removal kit
+	[7907] = {{1754, 1755}, {1760, 1761}}, -- removal kit
 	[20252] = {{20197, 20198}, {20199, 20200}} -- canopy kit
 }
 
-local function internalBedTransform(item, targetItem, toPosition, ids)
-	targetItem:transform(ids[1])
+local function internalBedTransform(item, targetItem, tile, toPosition, itemArray)
```
Comment (marksamman) [inline data/actions/scripts/other/bed_modification_kits.lua]: Why did you add the tile argument here?

**Comment by marksamman** [comment]
The problem isn't solely that you've squashed new commits, but also that you've kept adding new changes after initially opening the PR. If you hadn't squashed your commits, it would have been easier to know what you had changed after opening the pull request, and if you hadn't added new changes it wouldn't have mattered.

I disagree with your definition of cleanup if you're calling everything except for surprisebag.lua a cleanup. There are changes that affect semantics in multiple other files.

Yes, using ipairs is slower, but not as slow as the page you're linking to suggests (it's 4 years old though, so maybe it was accurate that long ago). In my tests, it was 8% slower with LuaJIT, and 20% slower with Lua, but it's a matter of nanoseconds, so it doesn't matter, we go for readability in this case. It takes millions of iterations before it's a matter of milliseconds. Whatever you're going to do inside the loop is going to dominate the total time anyway. The last paragraph under "Other Considerations" on that page is on point.

**Comment by WibbenZ** [comment]
I know that and im sorry that I diden't think about it.
But from my view you had not even seen or checked the pull, so it was either adding more things to this one or add a new one.
I might have done 4 changes in total, so that would be 4 new pulls or 4 commits(what I should have done).

To a certain extend IMO, at the same time to me a cleanup is still cleaning up things, thing like "local area", missing new line in files, spaces to tabs etc etc
To me a clean up is not only things like indentation etc

Sure np, ill revert it.

File: data/actions/scripts/other/bed_modification_kits.lua (hunk 11 above)
Comment (WibbenZ) [inline data/actions/scripts/other/bed_modification_kits.lua]: That was because I used the old tile, but Ninja pointed out that the position is changed. Forgot to change that when I reverted the tile userdata itself.

File: data/npc/scripts/The Oracle.lua [hunk 12]
```diff
@@ -64,13 +68,14 @@ local function creatureSayCallback(cid, type, msg)
...
 			npcHandler:say("SO BE IT!", cid)
-			doPlayerSetVocation(cid, vocation[cid])
-			doPlayerSetTown(cid, town[cid])
 			local destination = destination[cid]
 			npcHandler:releaseFocus(cid)
-			doSendMagicEffect(getCreaturePosition(cid), CONST_ME_TELEPORT)
-			doTeleportThing(cid, destination)
-			doSendMagicEffect(destination, CONST_ME_TELEPORT)
+
+			player:setVocation(Vocation(vocation[cid]))
```
Comment (marksamman) [inline data/npc/scripts/The Oracle.lua]: vocation and town must be set before releasing focus

File: data/npc/scripts/The Oracle.lua [hunk 13]
```diff
@@ -6,20 +6,21 @@ local vocation = {}
 local town = {}
 local destination = {}
 
-function onCreatureAppear(cid) npcHandler:onCreatureAppear(cid) end
-function onCreatureDisappear(cid) npcHandler:onCreatureDisappear(cid) end
-function onCreatureSay(cid, type, msg) npcHandler:onCreatureSay(cid, type, msg) end
-function onThink() npcHandler:onThink() end
+function onCreatureAppear(cid)				npcHandler:onCreatureAppear(cid)			end
```
Comment (marksamman) [inline data/npc/scripts/The Oracle.lua]: Either revert this or use spaces instead of tabs for alignment.

File: data/npc/scripts/runes.lua [hunk 14]
```diff
@@ -94,34 +93,40 @@ shopModule:addSellableItem({'underworld rod', 'underworld'}, 8910, 11000, 'under
...
-				talkState[talkUser] = 1
+	local player = Player(cid)
+	local items = {
+		[1] = 2190,
+		[2] = 2182,
+		[5] = 2190,
+		[6] = 2182
+	}
+
+	if msgcontains(msg, 'first rod') or msgcontains(msg, 'first wand')) then
```
Comment (marksamman) [inline data/npc/scripts/runes.lua]: Remove the last closing parenthesis.

Comment (WibbenZ) [inline data/npc/scripts/runes.lua]: Fixed

File: data/npc/scripts/The Oracle.lua [hunk 15]
```diff
@@ -29,13 +30,16 @@ end
 local function creatureSayCallback(cid, type, msg)
 	if not npcHandler:isFocused(cid) then
 		return false
-	elseif msgcontains(msg, "yes") and npcHandler.topic[cid] == 0 then
+	end
+
+	local player = Player(cid)
```
Comment (marksamman) [inline data/npc/scripts/The Oracle.lua]: This should be moved into the scope where it's actually used.

**Comment by marksamman** [comment]
You can squash your commits now.

**Comment by WibbenZ** [comment]
Done and ill remember to not squash before you have told me :p

**Comment by marksamman** [comment]
Thanks!
